src/multirust-cli/proxy_mode.rs

```diff
@@ -0,0 +1,28 @@
+use common::{run_inner, set_globals};
+use multirust::{Cfg, Result, Error};
+use multirust_utils::utils;
+use std::env;
+use std::path::PathBuf;
+
+pub fn main() -> Result<()> {
+    try!(::self_update::cleanup_self_updater());
```

Comment:
This'll be executed for all rustc/cargo invocations, right? Along those lines we probably don't want to do *too* much work in the setup here. I doubt this does much, however, but curious what you think?